src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/V2FeedParser.cs

```diff
@@ -158,6 +160,7 @@ public V2FeedParser(HttpSource httpSource, PackageSource source)
         /// </summary>
         public async Task<IReadOnlyList<V2FeedPackageInfo>> FindPackagesByIdAsync(
             string id,
+            V2FeedCacheContext cacheContext,
```

Comment:
How about throwing if this is null and adding a SourceCacheContext.NoCache static property that returns a cache context that contains options to disable caching (same as null does here)